test/inspect-test.js

```diff
@@ -4,368 +4,554 @@ const tape = require('tape');
+
+      sess.linesUntil(/}>/, (err, lines) => {
+        if (err) return cb(err);
+        lines = lines.join('\n');
+        t.ok(lines.includes('<Array: length=6'),
+            'length of hashmap.array should be 6');
+        cb(null);
+      });
+    }
+  },
+  // .long-array=0x000003df9cbe7aa9:<Array: length=20>,
+  'long-array': {
+    re: /.long-array=(0x[0-9a-f]+):<Array: length=20>/,
+    desc: '.long-array JSArray property',
+    validator(t, sess, addresses, name, cb) {
+      const address = addresses[name];
+      sess.send(`v8 inspect --array-length 10 ${address}`);
+
+      sess.linesUntil(/}>/, (err, lines) => {
+        if (err) return cb(err);
+        lines = lines.join('\n');
+        t.ok(lines.includes('<Array: length=20'),
+            'length of hashmap.long-array should be 6');
+        t.ok(/\[9\]=<Smi: 5>}>$/.test(lines),
+            'hashmap.long-array should have the right content');
+        cb(null);
+      });
+    }
+  },
+  // .array-buffer=0x000003df9cbe7df1:<ArrayBuffer: backingStore=0x00000000022509b0, byteLength=5>,
+  'array-buffer': {
+    re: new RegExp('.array-buffer=(0x[0-9a-f]+):' +
+                    '<ArrayBuffer: backingStore=0x[0-9a-f]+, byteLength=5>'),
+    desc: '.array-buffer JSArrayBuffer property',
+    validators: [(t, sess, addresses, name, cb) => {
+      const address = addresses[name];
+      sess.send(`v8 inspect ${address}`);
+
+      sess.linesUntil(/\]>/, (err, lines) => {
+        if (err) return cb(err);
+        lines = lines.join('\n');
+        const re = new RegExp(
+            '0x[0-9a-f]+:' +
+            '<ArrayBuffer: backingStore=0x[0-9a-f]+, byteLength=5: \\[\n' +
+            '  01, 02, 03, 04, 05\n' +
+            ']>');
+        t.ok(re.test(lines),
+            'hashmap.array-buffer should have the right content');
+        cb(null);
+      });
+    }, (t, sess, addresses, name, cb) => {
+      const address = addresses[name];
+      sess.send(`v8 inspect --array-length 1 ${address}`);
+
+      sess.linesUntil(/\]>/, (err, lines) => {
+        if (err) return cb(err);
+        lines = lines.join('\n');
+        const re = new RegExp(
+            '0x[0-9a-f]+:' +
+            '<ArrayBuffer: backingStore=0x[0-9a-f]+, byteLength=5: \\[\n' +
+            '  01 ...\n' +
+            ']>');
+        t.ok(re.test(lines),
+            'hashmap.array-buffer should have the right content with ' +
+            '--array-length 1');
+        cb(null);
+      });
+    }]
+  },
+  // .uint8-array=0x0000393071133e59:<ArrayBufferView: backingStore=0x000000000195b230, byteOffset=0, byteLength=6>,
+  // OR
+  // .uint8-array=0x000003df9cbe7eb9:<ArrayBufferView [neutered]>,
+  'uint8-array': {
+    re: new RegExp('.uint8-array=(0x[0-9a-f]+):<ArrayBufferView: ' +
+                    'backingStore=0x[0-9a-f]+, byteOffset=\\d+, ' +
+                    'byteLength=6>'),
+    desc: '.uint8-array JSArrayBufferView property',
+    optional: 'might be neutered',
```

Comment:
This is where the test is configured to be skippable (if we could not find such property, the validator won't be run, a skip message with the reason specified in `optional` will be printed)

Reply:
Would it be possible to use an array of regexes and check if one matches?  This `.optional` property means the test can never fail.  That's undesirable, IMO.